pkg/migrate/util.go

```diff
@@ -0,0 +1,342 @@
+package migrate
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"sync"
+
+	"github.com/Axway/agent-sdk/pkg/apic"
+	v1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/api/v1"
+	mv1a "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/management/v1alpha1"
+	"github.com/Axway/agent-sdk/pkg/apic/provisioning"
+)
+
+const (
+	serviceName  = "service-name"
+	instanceName = "instance-name"
+	revisionName = "revision-name"
+)
+
+var apiserviceName string
+
+// UpdateService - gets a list of instances for the service and updates their request definitions.
+func UpdateService(ri *v1.ResourceInstance, m *MarketplaceMigration) error {
```

Comment:
I see you exported this func so it can be used in provisioning.go. I think you can achieve the same thing by leaving all these funcs on `MarketplaceMigration` and doing your type conversion on the interface to get the `MarketplaceMigration` type and then calling `mig.UpdateService`